### ISSUE TITLE
Could it search for models that use time series such as RNN

### ISSUE THREAD

**Issue by guijuzhejiang** [google/model_search#46]
hi,I want to know how to use it to search for LSTM and GRU models.

**Comment by profitablealgos** [comment]
Yes, but you have to implement your own data provider class which outputs 3D Tensor.
I have tried, but not managed to do it because of my limited skills. But if you look at the example of the custom image data class, it should be possible to do something similar for timeseries

**Comment by Xiaoping777** [comment]
I am trying to do so...

still struggling on implementing the 2d dataset structure.  I am using sub-word vec to present the sentence, the shape is (512, 768), currently got issue:
ValueError: Cannot create a tensor proto whose content is larger than 2GB

updates: 
(1) limits the dataset size as 10 to remove this issue at the moment.
(2) it looks like the RNN model output of each hidden unit of 512 time series , while I want to get the result of final output

**Comment by JoshuaMathew** [comment]
> I am trying to do so...
> 
> still struggling on implementing the 2d dataset structure. I am using sub-word vec to present the sentence, the shape is (512, 768), currently got issue:
> ValueError: Cannot create a tensor proto whose content is larger than 2GB
> 
> updates:
> (1) limits the dataset size as 10 to remove this issue at the moment.
> (2) it looks like the RNN model output of each hidden unit of 512 time series , while I want to get the result of final output

@Xiaoping777 Could you share your code?

**Comment by Xiaoping777** [comment]
> > I am trying to do so...
> > still struggling on implementing the 2d dataset structure. I am using sub-word vec to present the sentence, the shape is (512, 768), currently got issue:
> > ValueError: Cannot create a tensor proto whose content is larger than 2GB
> > updates:
> > (1) limits the dataset size as 10 to remove this issue at the moment.
> > (2) it looks like the RNN model output of each hidden unit of 512 time series , while I want to get the result of final output
> 
> @Xiaoping777 Could you share your code?

Sorry, I only implemented the DNN, and gave up the RNN (CNN) , maybe pickup later when I have some time.

**Comment by JoshuaMathew** [comment]
> > > I am trying to do so...
> > > still struggling on implementing the 2d dataset structure. I am using sub-word vec to present the sentence, the shape is (512, 768), currently got issue:
> > > ValueError: Cannot create a tensor proto whose content is larger than 2GB
> > > updates:
> > > (1) limits the dataset size as 10 to remove this issue at the moment.
> > > (2) it looks like the RNN model output of each hidden unit of 512 time series , while I want to get the result of final output
> > 
> > 
> > @Xiaoping777 Could you share your code?
> 
> Sorry, I only implemented the DNN, and gave up the RNN (CNN) , maybe pickup later when I have some time.

That's fine, can you still share the code that you have even even if doesn't give you the expected output. I'm also trying to work with a 2D dataset using but I haven't been able to get it running.
